docker-compose.yaml

```diff
@@ -15,10 +16,13 @@ services:
     volumes:
       - './api:/home/kernelci/api'
       - './test:/home/kernelci/test'
+      - './migrations:/home/kernelci/migrations'
     ports:
       - '${API_HOST_PORT:-8001}:8000'
     env_file:
       - '.env'
+    depends_on:
+      - db
```

Comment:
Some notes about that: https://docs.docker.com/compose/startup-order/

It kind of helps to add `depends_on` but it's not enough to guarantee the DB will be ready before the API service starts talking to it.

Reply:
Thanks for the link.
Looking at different wrapper scripts for this.

Reply:
Tried to use `healthcheck` and `service_health` for the db dependency.
It turned out that docker-compose 3 doesn't support it https://forums.docker.com/t/depends-on-version-3/29463

Reply:
I verified different wrapper scripts mentioned at https://docs.docker.com/compose/startup-order/ for the DB dependency issue.

1. wait-for-it (https://github.com/vishnubob/wait-for-it):
This script is verifying if the TCP port of DB service is open without connection to it using `netcat` utility.
I integrated this tool with API and it seems working fine.

2. wait4x (https://github.com/atkrad/wait4x):
This tool connects to DB and runs ping command.
Set it up locally and tried to use it. However, it seems not working for any requests including http, tcp, and mongodb. I tried to debug the issue but no error logs are there.

3. dockerize (https://github.com/powerman/dockerize):
It does the same thing as the `wait-for-it` script. I haven't tried it though.

Another way is to write a python script using `motor` driver and connect to mongodb to verify the service status (basically what `wait4x` meant to do). And run the script from api container before running the fastAPI server.
In my opinion, checking the open TCP port is not enough and a test connection should be made in order to check the DB status.

What do you suggest? @gctucker @mgalka